salt/consul/query_template.sls

```diff
@@ -18,3 +18,29 @@ create_query_template_for_nearest_service:
     - match_type: pcre
     - decode: True
     - raise_error: False
+
+create_query_template_for_ops_service:
+  http.query:
+    - name: 'http://consul.service.consul:8500/v1/query'
+    - method: POST
+    - data: >-
+        {
+          "Name": "operations",
+          "Service": {
+            "Failover": {
+              "Datacenters": [
+                "operations",
+                "operations-qa"
+              ]
+            },
+            "Service": "${match(1)}"
+          },
+          "Template": {
+            "Regexp": "^operations-(.*?)$",
```

Comment:
Is this meant to match against `operations-qa` and `operations`? Just wondering about the `-`

Reply:
This is intended to allow for DNS records of the form `operations-elasticsearch.query.consul` similar to our `nearest-elasticsearch.query.consul` prepared query. That allows us to avoid having to write conditionals in our pillar data for things like `elasticsearch.service.operations.consul` vs `elasticsearch.service.operations-qa.consul`.

Reply:
And consul will know which `operations` (prod or qa) to point to?

Reply:
The way that will work is that a given VPC will only be connected to the appropriate operations environment, meaning that the query will only be able to resolve from the one network. This saves us from having to embed that logic in our pillar data and push it to the Consul layer.

Reply:
Are there any safeguards in place in case the secgroups/routing gets screwed up? Might be overkill, but just thinking about potential issues

Reply:
In other words, the QA environments will be updated to peer their networks with operations-qa, meaning that the query will then start resolving for the fluentd/elasticsearch/etc. services from the operations-qa environment instead of the existing operations environment.

Reply:
As far as safeguards, it will try to resolve one DC and then fail over to the other if it can't reach it. It doesn't really handle whether the secgroups are configured properly, but that's why we're testing in QA first :)